### PR TITLE
fix(TCOMP-2049): revert Entry move interface to class

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2049

### CODE REVIEW

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java [hunk 1]
```diff
@@ -319,15 +353,90 @@ void testToBuilder() {
...
+                .withMetadata(true)
+                .build());
         final Schema schemaNew = builder.build();
         assertEquals(3, schemaNew.getMetadata().size());
         assertEquals(3, schemaNew.getEntries().size());
         assertEquals(6, schemaNew.getAllEntries().count());
         assertEquals("meta1,data1,meta2,data2,data3,meta3", getSchemaFields(schemaNew));
     }
 
+    @Test
```
Comment (clesaec) [inline component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:368]: sanitizeConnectionName method is entirely defined in component-api module, so, it should be mainly test in this module.

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java [hunk 2]
```diff
@@ -319,15 +353,90 @@ void testToBuilder() {
...
+            final String sanitize = Schema.sanitizeConnectionName(randomString);
+            Assertions.assertTrue(checkPattern.matcher(sanitize).matches(), "'" + sanitize + "' don't match");
+
+            final String sanitize2 = Schema.sanitizeConnectionName(sanitize);
+            Assertions.assertEquals(sanitize, sanitize2);
+        }
+    }
+
+    @Test
+    void testAvoidCollision() {
```
Comment (clesaec) [inline component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:408]: Same as test sanitize.

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java [hunk 3]
```diff
@@ -36,41 +44,34 @@
...
 
-    private final Schema.Entry meta2 = new Schema.Entry.Builder() //
+    private final Schema.Entry meta2 = new SchemaImpl.EntryImpl.BuilderImpl() //
             .withName("meta2") //
             .withType(Schema.Type.STRING) //
             .withMetadata(true) //
             .withNullable(true) //
             .build();
 
-    @Test
```
Comment (clesaec) [inline component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:66]: Why remove this unit test

**Review by clesaec — COMMENTED** [review]
Always little sad to see we need so boiler plate code just "in case" ... but, it's the game :) 
Just questions about some unit tests

File: component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java [hunk 4]
```diff
@@ -218,83 +256,4 @@ public Builder withNewSchema(final Schema schema) {
         Assertions.assertFalse(Schema.Type.DATETIME.isCompatible(10));
     }
 
-    @Test
-    void testSanitize() {
```
[the remark this answers is not in the document]
Comment (undx) [reply component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java:222]: It has been moved to `SchemaImplTest` for using real `Entry` implementation and tested method is now in SchemaImpl.

File: component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java [hunk 5]
```diff
@@ -218,83 +256,4 @@ public Builder withNewSchema(final Schema schema) {
...
-            final String sanitize = Schema.sanitizeConnectionName(randomString);
-            Assertions.assertTrue(checkPattern.matcher(sanitize).matches(), "'" + sanitize + "' don't match");
-
-            final String sanitize2 = Schema.sanitizeConnectionName(sanitize);
-            Assertions.assertEquals(sanitize, sanitize2);
-        }
-    }
-
-    @Test
-    void testAvoidCollision() {
```
[the remark this answers is not in the document]
Comment (undx) [reply component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java:261]: It has been moved to `SchemaImplTest` for using real `Entry` implementation and tested method is now in SchemaImpl.

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 1 above)
Comment (undx) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:368]: see above

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 2 above)
Comment (undx) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:408]: see above

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 3 above)
Comment (undx) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:66]: Lombok `@Getter` seems to make test failing w/ no implementation of `getName()`, etc. methods...

File: component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (clesaec) [reply component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java:222]: `static String sanitizeConnectionName(final String name)` does not use Schema.Entry interface ?

File: component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (clesaec) [reply component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java:261]: ok

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 1 above)
Comment (clesaec) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:368]: move it back to component-api, as it does not use Schema.Entry interface

File: component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (undx) [reply component-api/src/test/java/org/talend/sdk/component/api/record/SchemaTest.java:222]: oups, my bad!

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 1 above)
Comment (undx) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:365]: done

File: component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java (hunk 3 above)
Comment (clesaec) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:66]: try `@Getter(onMethod = @__({@Override}))`

Comment (undx) [reply component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java:66]:
```java.lang.AssertionError: EqualsVerifier found a problem in class org.talend.sdk.component.runtime.record.SchemaImpl.
-> Abstract delegation: SchemaImpl's hashCode method delegates to an abstract method:
 Receiver class org.talend.sdk.component.api.record.Entry$$DynamicSubclass$645482568 does not define or inherit an implementation of the resolved method 'abstract java.lang.String getName()' of interface org.talend.sdk.component.api.record.Schema$Entry.
```

**Review by clesaec — APPROVED** [review]
LGTM

**Pull request merged**